### PR TITLE
Proposal for standardized storage of installable artifacts

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
# What Does This Do

For internal tooling needs (e.g. System Tests), and for having consistent and reliable experience fetching the tracer artifacts. We can leverage a simplified approach of [OCI Artifacts](https://github.com/opencontainers/artifacts) to store the final build artifacts (e.g. dd-java-agent.jar) in GitHub Container Registry.

From GitLab we will generate the docker images with last release and last snapshot on every commit in master branch.

see also: https://docs.google.com/document/d/1HbMGgZFRnIYjfnXGJGyk0LCwWs2op12R0BE58wma448/edit#

# Motivation

Simplify the mechanisms  to download snapshots and releases of the tracer. Simplify tests automation.

# Additional Notes

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ variables:
     value: "false"
     description: "Set to true to override rules in the reliability-env pipeline (e.g. override 'only deploy master')"
 
+include:
+  remote: https://gitlab-templates.ddbuild.io/libdatadog/include/docker-base-image-authenticated.yml
```
Comment (pawelchcki) [inline .gitlab-ci.yml:17]: we could make the name include `ci_authenticated_job` to ensure ppl will understand where the `ci_authenticated_job` comes from

File: tooling/ci/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN touch -c -a -m -d @0 /binaries/*
+
+FROM scratch as collect
+COPY --from=java_agent_debug /binaries/dd-java-agent.jar /dd-java-agent.jar
+COPY --from=java_agent_debug /binaries/LIBRARY_VERSION /LIBRARY_VERSION
+COPY --from=java_agent_debug /binaries/LIBDDWAF_VERSION /LIBDDWAF_VERSION
+COPY --from=java_agent_debug /binaries/APPSEC_EVENT_RULES_VERSION /APPSEC_EVENT_RULES_VERSION
+
+FROM scratch
+COPY --from=collect /* /
```
Comment (PerfectSlayer) [inline tooling/ci/Dockerfile:20]: What is the rationale behind using two more stages from scratch?
Can't we just use one like this:
```
FROM scratch
COPY --from=java_agent_debug /binaries/dd-java-agent.jar /binaries/LIBRARY_VERSION /binaries/LIBDDWAF_VERSION /binaries/APPSEC_EVENT_RULES_VERSION /
```

Comment (robertomonteromiguel) [reply tooling/ci/Dockerfile]: you are right. Changed! Thanks!

**Review by PerfectSlayer — APPROVED** [review]
LGTM for the tracer part.
Won't judge for the whole OCI artifact initiative :)

**Review by pawelchcki — APPROVED** [review]
LGTM too

@PerfectSlayer 
>Won't judge for the whole OCI artifact initiative :)
At this point we're POC'ing it - to see how much it improves our workflow etc. We'll see :man_shrugging:

**Pull request merged**